kafka_consumer/datadog_checks/kafka_consumer/new_kafka_consumer.py

```diff
@@ -283,6 +285,8 @@ def _report_consumer_offsets_and_lag(self, contexts_limit):
 
                 if reported_contexts >= contexts_limit:
                     continue
+                if not self._data_streams_enabled:
+                    continue
```

Comment:
It looks like tests are failing because we still expect the `consumer_lag` metric to be submitted in all cases. Can you update the tests?

Reply:
I added a test case specific when the flag is enabled. What do you think?

Reply:
Could you also add a test to assert the metric is not there when it's disabled?